R/nneo_wrangle.R

```diff
@@ -108,20 +111,57 @@ nneo_wrangle<-function(site_code="BART",time_start="2016-06-20",time_end=NULL,
     searh_terms <- paste0(time_agr, "min")
     files_time_agr <- sort(files_package$url[grep(searh_terms,
                                                  files_package$url)])
+
+
     #get the data using nneo_file:
     # data_all <- lapply(year_month, function(y) {
     #     lapply(unique(files_time_agr), function(x) {
-    #         nneo::nneo_file(product_code = substr(x, 15, 27), 
+    #         nneo::nneo_file(product_code = substr(x, 15, 27),
     #             site_code = site_code, year_month = y, filename = x)
     #     })
     # })
     data_all <- lapply(files_time_agr, nGET2, ...)
     data_all <- lapply(data_all, function(z) {
         data.table::fread(z, stringsAsFactors = FALSE, data.table = FALSE)
     })
-    data_merge <- tibble::as_tibble(
-        data.table::setDF(data.table::rbindlist(data_all, fill=TRUE, use.names=TRUE))
-    )
+
+    #keep all spatial levels (bug fix)
+    spatial_search<-"\\.\\d{3}\\.\\d{3}\\.\\d{3}"
+    #grep(spatial_terms,files_time_agr[1])
+    spatial_terms<-sub("\\.\\d{3}\\.","",stringr::str_extract(files_time_agr,spatial_search))
+    #assign spatial terms to names:
+    names(data_all)<-spatial_terms
+    #append spatial terms to col names:
+    for(i in 1:length(data_all)){
+            names(data_all[[i]])<-c(names(data_all[[i]][,1:2]),
+                                    paste0(names(data_all[[i]][3:length(data_all[[i]])]),".",spatial_terms[i]))
+            #set row.names to NULL:
+            row.names(data_all[[i]])<-NULL
+    }
+
+    # #create 'true' time sequence to merge to (in case of missing time stamps)
```

Comment:
@jaroberti can we remove all this commented out code starting here? or do you want to save for something?